compiler/ballerina-lang/src/main/java/io/ballerina/projects/ModuleContext.java

```diff
@@ -556,6 +558,15 @@ ModuleContext duplicate(Project project) {
                 this.moduleDescDependencies, this.resourceContextMap, this.testResourceContextMap);
     }
 
+    private Collection<DocumentId> sortDocuments(Map<DocumentId, DocumentContext> docContextMap) {
```

Comment:
We create a new ModuleContext for every document edit. I think this would affect the LS since it will sort for every edit request. IMO, we can specifically sort the map when we create a package and when a document is added. @sameerajayasoma WDYT?

Reply:
Addressed this in db53c289c2f4ec33ec8d539e58b4877ffdf0a0f7